File: CatalogoVeiculos_Web/src/app/Models/Modelo.ts
import { Marca } from "./Marca";

export class Modelo{
    modeloId: number = 0;
    nomeModelo: string = '';
    statusModelo: boolean = true;
    marcaId: number = 0;

    marca: Marca = {
        marcaId: 0,
        nomeMarca: '',
        statusMarca: true
    };
    marcas?: Marca[];
}

File: CatalagoVeiculos/CatalogoVeiculos.Domain/Entities/Modelo.cs
namespace CatalogoVeiculos.Domain.Entities
{
    public class Modelo
    {
        public int ModeloId { get; set; }
        public string NomeModelo { get; set; }
        public int MarcaId { get; set; }
        public bool StatusModelo { get; set; }
        public virtual Marca Marca { get; set; }
    }
}
